docs/DeveloperGuide.md

```diff
@@ -324,13 +329,149 @@ Below are instructions to perform manual testing of the application. Please refe
+2. Assumption: You have a module `CS2113T`.
+3. Test Case: `grade CS2113T A+` <br>
+   Expected: The grade of `CS2113T` has been set to `A+`.
+4. Test Case: `grade CS2113T E` <br>
+   Expected: No grade is set. Error details in the message shows that `E` is an invalid module grade.
+
+<br>
+
+### Calculating GPA
+1. Prerequisite: There are existing modules in the application.
+2. Assumption: You have a module `CS2113T` of `4` modular credits and grade `A+` and a module `CS2101` of `4` modular credits and grade `B`.
+3. Test Case: `gpa` <br>
+   Expected: Your `gpa` is calculated as `4.25`.
+
+<br>
+
+### Showing Help
+1. Test Case: `help` <br>
+   Expected: A message for format of the `help` command is shown.
+2. Test Case: `help add"` <br>
+   Expected: A message for the format of the `add` command is shown.
 
+<br>
+
+### Setting options
+1. Test Case: `option` <br>
+   Expected: Available configuration settings is shown with its corresponding value.
+2. Test Case: `option INVALID_CONFIG` <br>
+   Expected: An error message is shown detailing that there is no configuration called `INVALID_CONFIG`.
+3. Test Case: `option SHOW_COMPLETED_TASKS=invalid` <br>
+   Expected: No configuration is changed. Error details in the message shows that the value `invalid` is not supported for configuration `SHOW_COMPLETED_TASKS`.
+
+<br>
+
+### Adding tags to a task
+1. Prerequisite: There are existing tasks in the application.
+2. Assumption: You have at least one task in `General Tasks`.
+3. Test Case: `tag add 1 IMPT` <br>
+   Expected: The tag `IMPT` is added to your first task in `General Tasks`.
+4. Test Case: `tag add 1 .invalid` <br>
+   Expected: No tag is added. Error details in the message shows that `.invalid` is an invalid tag name.
+
+<br>
+
+### Deleting tags to a task
+1. Prerequisite: There are tasks with tags in the application.
+2. Assumption: The first task in `General Tasks` is tagged as `IMPT` only.
+3. Test Case: `tag del 1 IMPT` <br>
+   Expected: The tag `IMPT` is removed from the first task in `General Tasks`.
+4. Test Case: `tag del 1 OTHERS` <br>
+   Expected: No tag is deleted. Error details in the message shows that no such tag exists
+
+<br>
+
+### Listing all modules and tasks
+1. Prerequisite: There are tasks that were tagged.
+2. Assumption: There are tasks that were tagged as `IMPT`.
+3. Test Case: `list` <br>
+   Expected: All of your modules and tasks are shown.
+4. Test Case: `list IMPT` <br>
+   Expected: Only tasks tagged as `IMPT` are shown. All of your modules are still shown regardless.
+
+<br>
+
+### Marking and unmarking a task as completed
+1. Prerequisite: There are existing tasks in the application.
+2. Assumption: There are at least one task in `General Tasks`.
+3. Test Case: `mark c 1` <br>
+   Expected: The first task in `General Tasks` is marked as completed.
+4. Test Case: `mark u 1` <br>
+   Expected: The first task in `General Tasks` is unmarked.
```

Comment:
Maybe `marked as uncompleted` instead of `unmarked`.

Reply:
yup changed